shell/src/main/java/alluxio/cli/fsadmin/command/report/SummaryCommand.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.fsadmin.command.report;
+
+import alluxio.client.block.RetryHandlingBlockMasterClient;
+import alluxio.client.MetaMasterClient;
+import alluxio.client.RetryHandlingMetaMasterClient;
+import alluxio.exception.status.UnavailableException;
+import alluxio.master.MasterClientConfig;
+import alluxio.util.CommonUtils;
+import alluxio.util.FormatUtils;
+import alluxio.wire.BlockMasterInfo;
+import alluxio.wire.BlockMasterInfo.BlockMasterInfoField;
+import alluxio.wire.MasterInfo;
+import alluxio.wire.MasterInfo.MasterInfoField;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Print Alluxio cluster summarized information.
+ * This class depends on meta master client and block master client.
+ */
+public class SummaryCommand {
+  /** Print Alluxio cluster summarized information. */
+  public static void printSummary() {
+    System.out.println("Alluxio Cluster Summary: ");
+
+    try (MetaMasterClient client =
+        new RetryHandlingMetaMasterClient(MasterClientConfig.defaults())) {
+      Set<MasterInfoField> masterInfoFilter = new HashSet<>((Arrays
+          .asList(MasterInfoField.MASTER_ADDRESS, MasterInfoField.WEB_PORT,
+              MasterInfoField.RPC_PORT, MasterInfoField.START_TIME_MS,
+              MasterInfoField.UP_TIME_MS, MasterInfoField.VERSION,
+              MasterInfoField.SAFE_MODE)));
+      MasterInfo masterInfo = client.getMasterInfo(masterInfoFilter);
+
+      System.out.println("    Master Address: " + masterInfo.getMasterAddress());
+      System.out.println("    Web Port: " + masterInfo.getWebPort());
+      System.out.println("    Rpc Port: " + masterInfo.getRpcPort());
+      System.out.println("    Started: "
+          + CommonUtils.convertMsToDate(masterInfo.getStartTimeMs()));
+      System.out.println("    Uptime: "
+          + CommonUtils.convertMsToClockTime(masterInfo.getUpTimeMs()));
+      System.out.println("    Version: " + masterInfo.getVersion());
+      System.out.println("    Safe Mode: " + masterInfo.isSafeMode());
+    } catch (UnavailableException e) {
+      e.printStackTrace();
```

Comment:
other shell commands usually propagate exceptions so that they can be handled the same way at the top-level shell handler

Reply:
In summary command
    try (MetaMasterClient client =
             new RetryHandlingMetaMasterClient(MasterClientConfig.defaults())) {
      ....
      MasterInfo masterInfo = client.getMasterInfo(masterInfoFilter);
      if (masterInfo == null) {
        throw new IOException("Cannot get meta master info from meta master client");
      }
      .....
    } catch (UnavailableException e) {
      e.printStackTrace();
      System.out.println("Please check your Alluxio master status.");
    }

I think for AlluxioException and IOEexception, just throw them is a good choice.
For unavailableException, I need to catch it so maybe just print error and help message is better.

Reply:
why do we need to catch it? We could declare it to be thrown instead

Reply:
Changed to throw IOException. Thanks